packages/liveblocks-react-ui/src/primitives/Markdown.tsx

```diff
@@ -168,47 +248,29 @@ export function BlockTokenComp({
               />
             );
           })}
-        </blockquote>
+        </Blockquote>
       );
     }
     case "html": {
       return token.text;
     }
     case "heading": {
-      let HeadingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
-      switch (token.depth) {
-        case 1:
-          HeadingTag = "h1";
-          break;
-        case 2:
-          HeadingTag = "h2";
-          break;
-        case 3:
-          HeadingTag = "h3";
-          break;
-        case 4:
-          HeadingTag = "h4";
-          break;
-        case 5:
-          HeadingTag = "h5";
-          break;
-        case 6:
-          HeadingTag = "h6";
-          break;
-        default:
-          HeadingTag = "h1";
-          break;
+      const Heading = components?.Heading ?? defaultComponents.Heading;
+
+      if (token.depth < 1 || token.depth > 6) {
+        return null;
       }
+
       return (
-        <HeadingTag>
+        <Heading level={token.depth as 1 | 2 | 3 | 4 | 5 | 6}>
```

Comment:
I didn't want to do a `as`, so I used a `switch-case` statement, but I am also okay with this as it might be more pragmatic.